ledger/storage/db.go

```diff
@@ -83,27 +83,26 @@ func setOptions(o *badger.Options) *badger.Options {
 
 // NewDB returns storage.DB with BadgerDB instance initialized by opts.
 // Creates database in provided dir or in current directory if dir parameter is empty.
-func NewDB(conf configuration.Ledger, opts *badger.Options) (*DB, error) {
+func NewDB(conf configuration.Ledger, opts *badger.Options) *DB {
 	opts = setOptions(opts)
 	dir, err := filepath.Abs(conf.Storage.DataDirectory)
 	if err != nil {
-		return nil, err
+		panic(err)
```

Comment:
It's not a go way. at all.

Reply:
https://golang.org/doc/effective_go.html#panic

The execution of the whole binary cannot continue without it. It's pointless to check this error.

Reply:
panic or nice error processing (which should be in priority) is caller responsibility